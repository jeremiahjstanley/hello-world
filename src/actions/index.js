export const setLocation = (location) => ({
  type: 'SET_LOCATION',
  location
});

export const setDataSet = (dataSet) => ({
  type: 'SET_DATASET',
  dataSet
});

export const clearDataSet = () => ({
  type: 'CLEAR_DATASET',
});

export const setDataBase = (dataBase) => ({
  type: 'SET_DATABASE',
  dataBase
});

export const fetchLocationDataSuccess = (locationData) => ({
  type: 'FETCH_LOCATION_DATA_SUCCESS',
  locationData
});

export const hasErrored = (status) => ({
  type: 'HAS_ERRORED',
  hasErrored: status
});

export const madeSearch = (status) => ({
  type: 'MADE_SEARCH',
  madeSearch: status
});


