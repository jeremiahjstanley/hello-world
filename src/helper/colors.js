export const colors = [
'#e700ff',
'#0081ff',
'#ff0063',
'#74d2e7',
'#00f1ff',
'#fd5c63',
'#48a9c5',
'#0085ad',
'#8db9ca',
'#4298b5',
'#00205b',
'#d20962',
'#f47721',
'#00a78e',
'#7d3f98',
'#037ef3',
'#279b37',
'#008374',
'#0099e5',
'#ff4c4c',
'#34bf49'
]