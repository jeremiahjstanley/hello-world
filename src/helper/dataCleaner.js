export const goveranceIndicatorCleaner = (estimates, numberOfSources, percentileRank, standardError) => {
  const cleanEstimates = estimates.dataset.data.map(dataPoint => {
    return {"name": dataPoint[0], "Estimate": dataPoint[1]}
  });
  const cleanNumberOfSources = numberOfSources.dataset.data.map((dataPoint, index) => {
    return {...cleanEstimates[index], "name": dataPoint[0], "Number of Sources": dataPoint[1]}
  })
  const cleanPercentileRank = percentileRank.dataset.data.map((dataPoint, index) => {
    return {...cleanNumberOfSources[index], "name": dataPoint[0], "Percentile Rank": dataPoint[1]}
  })
  const cleanStandardError = standardError.dataset.data.map((dataPoint, index) => {
    return {...cleanPercentileRank[index], "name": dataPoint[0], "Standard Error": dataPoint[1]}
  })
  return cleanStandardError
}