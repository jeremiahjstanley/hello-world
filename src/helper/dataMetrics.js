export const dataMetrics = [
  {
    name: 'Voice and Accountability', 
    dataset_code: 'VA', 
    database_code: 'WWGI'
  },
  {
    name: 'Political Stability and Absence of Violence',
    dataset_code: 'PV',
    database_code: 'WWGI'
  },
  {
    name: 'Government Effectiveness',
    dataset_code: 'GE',
    database_code: 'WWGI'},
  {
    name: 'Regulatory Quality',
    dataset_code: 'RQ',
    database_code: 'WWGI'
  },
  {
    name: 'Rule of Law',
    dataset_code: 'RL',
    database_code: 'WWGI'
  },
  {
    name: 'Control of Corruption', 
    dataset_code: 'CC',
    database_code: 'WWGI'},
]
