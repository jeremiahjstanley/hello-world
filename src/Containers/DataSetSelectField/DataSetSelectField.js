import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setDataSet } from '../../actions';
import { dataMetrics } from '../../helper/dataMetrics';

export class DataSetSelectField extends Component {
  constructor() {
    super()

    this.state = {
      dataSet: dataMetrics[0].datasets[0]
    }
  }

  componentDidMount() {
    this.props.selectDataSet(this.state.dataSet)
  }

  selectMetric = (event) => {
    const dataSet = this.props.dataSet.find(metric => event.target.value === metric.name);
    this.props.selectDataSet({...dataSet})
  }

  render() {
    // const options = this.props.dataSet.map((metric, index) => (
    //   <option key={index}>
    //     { metric.name }
    //   </option>
    // ))
    const options = 'cat'
    return (
      <select onChange={this.selectMetric}>
        { options }
      </select>
    )
  };
}

export const mapStateToProps = (state) => ({
  dataSet: state.dataBase.datasets 
})

export const mapDispatchToProps = (dispatch) => ({
  selectDataSet: (dataSet) => dispatch(setDataSet(dataSet))
})

export default connect(mapStateToProps, mapDispatchToProps)(DataSetSelectField);


