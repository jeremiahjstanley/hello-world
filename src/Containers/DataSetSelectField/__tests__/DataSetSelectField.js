import React from 'react';
import { shallow } from 'enzyme';
import { DataSetSelectField, mapStateToProps, mapDispatchToProps } from '../DataSetSelectField';
import { setDataSet } from '../../../actions';
import { dataMetrics } from '../../../helper/dataMetrics';


describe('DataSetSelectField', () => {

  let wrapper;
  let mockEvent;
  let mockSelectDataSet;
  let mockDataBase;
  let mockDataSet; 

  beforeEach(() => {

    mockDataBase = 'Worldwide Governance Indicators';

  	mockDataSet = {
      name: 'Voice and Accountability', 
      dataset_code: 'VA'
    };

  	mockSelectDataSet = jest.fn();

    wrapper = shallow(
    	<DataSetSelectField 
    		dataBase={mockDataBase}
    		selectDataSet={mockSelectDataSet}
    	/>
  	);

  });

	it('should match the snapshot when the state has a dataBase property', () => {

		expect(wrapper).toMatchSnapshot();

	});

	it('should match the snapshot when the state has does not have a dataBase property', () => {

    wrapper = shallow(
    	<DataSetSelectField 
    		dataBase={ undefined }
    		selectDataSet={mockSelectDataSet}
    	/>
  	);

		expect(wrapper).toMatchSnapshot();

	});

  it('should call selectDataSet when the dropdown is has a change event', () => {

  	mockEvent = { target: { value: 'Voice and Accountability'} };

    wrapper.instance().selectMetric(mockEvent);

    expect(mockSelectDataSet).toHaveBeenCalledWith(mockDataSet);

  });

  it('should derive local state from props using  getDerivedStateFromProps', () => {

		const mockProps = {
			database: {
		    name: 'Worldwide Governance Indicators',
		    database_code: 'WWGI',
		    datasets: [
		      {
		        name: 'Voice and Accountability', 
		        dataset_code: 'VA'
		      
		      },
		      {
		        name: 'Political Stability and Absence of Violence',
		        dataset_code: 'PV'
		      
		      },
		      {
		        name: 'Government Effectiveness',
		        dataset_code: 'GE'
		      },
		      {
		        name: 'Regulatory Quality',
		        dataset_code: 'RQ'
		      
		      },
		      {
		        name: 'Rule of Law',
		        dataset_code: 'RL'
		      
		      },
		      {
		        name: 'Control of Corruption', 
		        dataset_code: 'CC'
		      }
		    ]
	  	}
	  };

		const mockState = {
			dataBase: {
		    name: 'Worldwide Governance Indicators',
		    database_code: 'WWGI',
		    datasets: [
		      {
		        name: 'Voice and Accountability', 
		        dataset_code: 'VA'
		      
		      },
		      {
		        name: 'Political Stability and Absence of Violence',
		        dataset_code: 'PV'
		      
		      },
		      {
		        name: 'Government Effectiveness',
		        dataset_code: 'GE'
		      },
		      {
		        name: 'Regulatory Quality',
		        dataset_code: 'RQ'
		      
		      },
		      {
		        name: 'Rule of Law',
		        dataset_code: 'RL'
		      
		      },
		      {
		        name: 'Control of Corruption', 
		        dataset_code: 'CC'
		      }
		    ]
	  	}
	  };
		
		wrapper.setProps(mockProps);

		const result = wrapper.state();

		expect(result).toEqual(mockState);

  });

	describe('mapStateToProps', () => {

		let actionToDispatch;
		let mockState;
		let mockDispatch;

		beforeEach(() => {

	    mockState = {
	  		dataBase: {
		  		name: 'Worldwide Governance Indicators',
		  		database_code: 'WWGI'	
	  		}
	  	};

		});

		it('should return an object with dataBase key', () => {
			
			const expected = {
				dataBase: mockDataBase
			};

			const mappedProps = mapStateToProps(mockState);

			expect(mappedProps).toEqual(expected);

		});
		
	});

	describe('mapDispatchToProps', () => {

		let actionToDispatch;
		let mockDispatch;

		beforeEach(() => {

			mockDispatch = jest.fn();

		});

		it('calls dispatch with the setDataBase action when called', () => {

			const mappedProps = mapDispatchToProps(mockDispatch);

			actionToDispatch = setDataSet(mockDataSet);

			mappedProps.selectDataSet(mockDataSet);

			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);

		});
		
	});
	
});