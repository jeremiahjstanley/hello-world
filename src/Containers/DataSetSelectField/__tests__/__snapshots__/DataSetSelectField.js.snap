// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSetSelectField should match the snapshot when the state has a dataBase property 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataSetSelectField
    dataBase="Worldwide Governance Indicators"
    selectDataSet={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option>
          Voice and Accountability
</option>,
        <option>
          Political Stability and Absence of Violence
</option>,
        <option>
          Government Effectiveness
</option>,
        <option>
          Regulatory Quality
</option>,
        <option>
          Rule of Law
</option>,
        <option>
          Control of Corruption
</option>,
      ],
      "onChange": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Voice and Accountability",
        },
        "ref": null,
        "rendered": "Voice and Accountability",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Political Stability and Absence of Violence",
        },
        "ref": null,
        "rendered": "Political Stability and Absence of Violence",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Government Effectiveness",
        },
        "ref": null,
        "rendered": "Government Effectiveness",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "Regulatory Quality",
        },
        "ref": null,
        "rendered": "Regulatory Quality",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "Rule of Law",
        },
        "ref": null,
        "rendered": "Rule of Law",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": "Control of Corruption",
        },
        "ref": null,
        "rendered": "Control of Corruption",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Voice and Accountability
</option>,
          <option>
            Political Stability and Absence of Violence
</option>,
          <option>
            Government Effectiveness
</option>,
          <option>
            Regulatory Quality
</option>,
          <option>
            Rule of Law
</option>,
          <option>
            Control of Corruption
</option>,
        ],
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Voice and Accountability",
          },
          "ref": null,
          "rendered": "Voice and Accountability",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Political Stability and Absence of Violence",
          },
          "ref": null,
          "rendered": "Political Stability and Absence of Violence",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Government Effectiveness",
          },
          "ref": null,
          "rendered": "Government Effectiveness",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "Regulatory Quality",
          },
          "ref": null,
          "rendered": "Regulatory Quality",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "Rule of Law",
          },
          "ref": null,
          "rendered": "Rule of Law",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": "Control of Corruption",
          },
          "ref": null,
          "rendered": "Control of Corruption",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DataSetSelectField should match the snapshot when the state has does not have a dataBase property 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataSetSelectField
    dataBase={undefined}
    selectDataSet={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
      "className": "loading-text",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
        "className": "loading-text",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
